### PR TITLE
etcdutil: fix load range option

### LINKED ISSUE #7689: PD continues sending merge operator to TiKV and then time out
## Bug Report

When we enable the scheduling service, PD wants to merge the region 302944954 and 302944946.
Here is the rule fit for these two regions:
```
sh-5.1# curl http://127.0.0.1:2379/pd/api/v1/config/rules/region/302944954/detail
{
    "rule-fits": [
        {
            "rule": {
                "group_id": "pd",
                "id": "default",
                "start_key": "",
                "end_key": "",
                "role": "voter",
                "is_witness": false,
                "count": 3,
                "location_labels": [
                    "topology.kubernetes.io/region",
                    "topology.kubernetes.io/zone",
                    "serverless.tidbcloud.com/partition",
                    "kubernetes.io/hostname"
                ],
                "version": 1
            },
            "peers": [
                {
                    "id": 452329816,
                    "store_id": 452861088
                },
                {
                    "id": 452419226,
                    "store_id": 452861087
                },
                {
                    "id": 460931485,
                    "store_id": 452861086
                }
            ],
            "peers-different-role": null,
            "isolation-score": 300,
            "witness-score": 0
        }
    ],
    "orphan-peers": null
}
sh-5.1# curl http://127.0.0.1:2379/pd/api/v1/config/rules/region/302944946/detail
{
    "rule-fits": [
        {
            "rule": {
                "group_id": "pd",
                "id": "default",
                "start_key": "",
                "end_key": "",
                "role": "voter",
                "is_witness": false,
                "count": 3,
                "location_labels": [
                    "topology.kubernetes.io/region",
                    "topology.kubernetes.io/zone",
                    "serverless.tidbcloud.com/partition",
                    "kubernetes.io/hostname"
                ],
                "version": 1
            },
            "peers": [
                {
                    "id": 452470008,
                    "store_id": 452861087
                },
                {
                    "id": 462779468,
                    "store_id": 452861088
                },
                {
                    "id": 462779469,
                    "store_id": 452861086
                }
            ],
            "peers-different-role": null,
            "isolation-score": 300,
            "witness-score": 0
        }
    ],
    "orphan-peers": null
}
```
But in TiKV side, these operator is rejected because the region belongs to two different keyspaces.
```
[WARN] [peer_fsm.rs:1002] ["failed to propose merge"] [err="KvEngine check merge shards from different keyspaces require to be empty ...]
```
More detailed information:
```
source: 
region id: 302944946 
start_key: 78009D2600000000FB 
end_key: 78009D2700000000FB

target: 
region id: 302944954 
start_key: 78009D2700000000FB 
end_key: 78009D2900000000FB

78009D2600000000FB -> keyspace id 40230
» keyspace show id 40230
{
    "id": 40230,
    "name": "DZ6h87arR3U4M33",
    "state": "ENABLED",
    "created_at": 1688637837,
    "state_changed_at": 1688637837,
    "config": {
        "gc_life_time": "6000",
        "safe_point_version": "v2",
        "serverless_cluster_id": "",
        "serverless_project_id": "",
        "serverless_tenant_id": "",
        "tso_keyspace_group_id": "4",
        "user_kind": "basic"
    }
}

78009D2700000000FB -> keyspace id 40231
» keyspace show id 40231
{
    "id": 40231,
    "name": "3Ker9q2AQjHxazx",
    "state": "TOMBSTONE",
    "created_at": 1688637837,
    "state_changed_at": 1689244223,
    "config": {
        "gc_life_time": "6000",
        "serverless_cluster_id": "e2e-test-1665-1555513219",
        "serverless_project_id": "",
        "serverless_tenant_id": "e2e-tenant-id",
        "tso_keyspace_group_id": "2",
        "user_kind": "basic"
    }
}

78009D2900000000FB -> keyspace id 40233
» keyspace show id 40233
{
    "id": 40233,
    "name": "48cDkthwxwKt9xx",
    "state": "ENABLED",
    "created_at": 1688637838,
    "state_changed_at": 1688637838,
    "config": {
        "gc_life_time": "6000",
        "safe_point_version": "v2",
        "serverless_cluster_id": "",
        "serverless_project_id": "",
        "serverless_tenant_id": "",
        "tso_keyspace_group_id": "4",
        "user_kind": "basic"
    }
}
```

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #7689

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/utils/etcdutil/etcdutil.go [hunk 1]
```diff
@@ -863,10 +863,9 @@ func (lw *LoopWatcher) watch(ctx context.Context, revision int64) (nextRevision
 	}
 }
 
-func (lw *LoopWatcher) load(ctx context.Context) (nextRevision int64, err error) {
-	ctx, cancel := context.WithTimeout(ctx, DefaultRequestTimeout)
-	defer cancel()
+func (lw *LoopWatcher) load() (nextRevision int64, err error) {
```
Comment (nolouch) [inline pkg/utils/etcdutil/etcdutil.go:866]: maybe we still need ctx to avoid load too long?

File: pkg/tso/keyspace_group_manager.go [hunk 2]
```diff
@@ -577,7 +571,7 @@ func (kgm *KeyspaceGroupManager) InitializeGroupWatchLoop() error {
 		putFn,
 		deleteFn,
 		postEventsFn,
-		clientv3.WithRange(endKey),
+		clientv3.WithPrefix(),
```
Comment (rleungx) [inline pkg/tso/keyspace_group_manager.go:574]: Shall we maintain these options in the same place?

File: pkg/utils/etcdutil/etcdutil_test.go [hunk 3]
```diff
@@ -537,6 +538,50 @@ func (suite *loopWatcherTestSuite) TestWatcherLoadLimit() {
...
+	re := suite.Require()
+	// use default limit to test 16384 key in etcd
+	count := 16384
+	ctx, cancel := context.WithCancel(suite.ctx)
+	defer cancel()
+	for i := 0; i < count; i++ {
+		suite.put(re, fmt.Sprintf("TestWatcherLoadLargeKey/test-%d", i), "")
+	}
+	cache := struct {
+		syncutil.RWMutex
```
Comment (rleungx) [inline pkg/utils/etcdutil/etcdutil_test.go:551]: Why do we need a lock here?

Comment (lhy1024) [reply pkg/utils/etcdutil/etcdutil_test.go:551]: we check result in a goroutine, watcher update cache in another goroutine

File: pkg/utils/etcdutil/etcdutil.go (hunk 1 above)
Comment (lhy1024) [reply pkg/utils/etcdutil/etcdutil.go]: ok, we use external ctx to control it

**Review by rleungx — APPROVED** [review]

File: pkg/utils/etcdutil/etcdutil.go [hunk 4]
```diff
@@ -689,12 +687,11 @@ func (lw *LoopWatcher) initFromEtcd(ctx context.Context) int64 {
...
 	)
-	ticker := time.NewTicker(defaultLoadFromEtcdRetryInterval)
+	ticker := time.NewTicker(defaultEtcdRetryInterval)
 	defer ticker.Stop()
-	ctx, cancel := context.WithTimeout(ctx, lw.loadTimeout)
-	defer cancel()
-
 	for i := 0; i < lw.loadRetryTimes; i++ {
+		ctx, cancel := context.WithTimeout(ctx, lw.loadTimeout)
+		defer cancel()
```
Comment (CabinfeverB) [inline pkg/utils/etcdutil/etcdutil.go:694]: put it at the end of loop?

**Review by CabinfeverB — COMMENTED** [review]
rest LGTM

File: pkg/utils/etcdutil/etcdutil.go (hunk 4 above)
Comment (lhy1024) [reply pkg/utils/etcdutil/etcdutil.go:694]: if without defer, it will need to be added in three places

**Pull request merged**